cmake/BLTPrivateMacros.cmake

```diff
@@ -212,24 +213,19 @@ macro(blt_setup_cuda_source_properties)
                             "${multiValueArgs}" ${ARGN} )
 
     # Check arguments
-
     if ( NOT DEFINED arg_BUILD_TARGET )
         message( FATAL_ERROR "Must provide a BUILD_TARGET argument to the 'blt_setup_cuda_source_properties' macro")
     endif()
 
-    
     if ( NOT DEFINED arg_TARGET_SOURCES )
         message( FATAL_ERROR "Must provide TARGET_SOURCES to the 'blt_setup_cuda_source_properties' macro")
     endif()
 
-
-    foreach (_file ${arg_TARGET_SOURCES})
-        if (${_file} MATCHES "\\.(f|F)\\*")
-            set(_non_cuda_sources ${_non_cuda_sources} ${_file})
-        else()
-            set(_cuda_sources ${_cuda_sources} ${_file})
-        endif()
-    endforeach()
+    set(_cuda_sources)
+    set(_non_cuda_sources)
+    blt_split_source_list_by_language(SOURCES      ${arg_TARGET_SOURCES}
```

Comment:
Should we also handle cuda files (``*.cu``)?

Reply:
Fixed.